### ISSUE TITLE
The precision recall curve (and maybe others) should returned a thresholds object with the same size as Xaxis and Yaxis

### ISSUE THREAD

**Issue by Yann-CV** [Lightning-AI/torchmetrics#2207]
## 🐛 Bug

The thresholds returned by the precision recall curve classes/functions (and maybe others) is missing the last value (threshold > max_prediction)

### To Reproduce

```
pr_curve = PrecisionRecallCurve(task="binary")
precision, recall, thresholds = pr_curve(pred, target)
assert precision.size() == recall.size()  == thresholds.size()  ---------> fails
```

### Expected behavior

```
pr_curve = PrecisionRecallCurve(task="binary")
precision, recall, thresholds = pr_curve(pred, target)
assert precision.size() == recall.size()  == thresholds.size()  ---------> does not fail
```

### Environment

TorchMetrics 1.2

### Additional context

This is an issue when you want to display the threshold value for all points in the curve (even though selecting this threshold makes no sense). It is also an issue when you want to create pandas.Dataframe from the returned tuple.

Quick fix on our side: we manually added a `max_prediction + epsilon `at the end of thresholds

**Comment by Borda** [comment]
seems as the same issue discovered by @matsumotosan in https://github.com/Lightning-AI/torchmetrics/pull/2154#issuecomment-1779318669

**Comment by SkafteNicki** [comment]
This is intended behavior as it is consistent with sklearn:
![image](https://github.com/Lightning-AI/torchmetrics/assets/24896311/c47b9d43-b375-4a54-a9fe-bc746a1d67b4)
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_curve.html

**Comment by Yann-CV** [comment]
@SkafteNicki sorry I did not noticed your answer... Then thanks.

If the point is to be sklearn compatible, then I guess it is a good point to follow their choice.

On my side, because I see Torchmetrics as a substitute to sklearn regarding metric computation, I still personally think this is a bad design choice.
